crates/sui-types/src/crypto.rs

```diff
@@ -1135,7 +1146,7 @@ pub struct AuthoritySignInfo {
 }
 impl AuthoritySignInfoTrait for AuthoritySignInfo {
     fn verify<T: Signable<Vec<u8>>>(&self, data: &T, committee: &Committee) -> SuiResult<()> {
```

Comment:
I know it's a temp change but it's kind of scary to leave here a function called `verify` with an implementation doesn't verify anything.

Reply:
Agreed. I'd like to see a pattern like we have with messages envelopes, something like:

```
struct VerifiedAuthoritySignInfo(AuthoritySignInfo);
impl VerifiedAuthoritySignInfo {
   fn from_sign_info(sig: AuthoritySignInfo) -> SuiResult<Self> {
      sig.verify()?;
      Ok(Self(sig))
   }
}
```

This way we can leverage the compiler to help us audit use of the unverified type.